system/View/View.php

```diff
@@ -356,7 +356,7 @@ public function setData(array $data = [], ?string $context = null): RendererInte
      */
     public function setVar(string $name, $value = null, ?string $context = null): RendererInterface
     {
-        if ($context) {
+        if ($context !== null && $context !== '') {
```

Comment:
cast to string

Reply:
Only `$context !== null` is sufficient.

Reply:
@kenjis it based on type declaration

```php
    public function setVar(string $name, $value = null, ?string $context = null): RendererInterface
    {
```

Reply:
`''` is impossible.
https://github.com/codeigniter4/CodeIgniter4/blob/e1c2f5d23b7234208f11b06504ed00119b48b3ac/system/View/View.php#L355

Reply:
@kenjis ok, updated to only check on null https://github.com/codeigniter4/CodeIgniter4/pull/7951/commits/15e91b24dcd955d91fbd2366d53ae41dc499f39e